huaweicloud/services/dds/resource_huaweicloud_dds_instance_parameters_modify.go

```diff
@@ -0,0 +1,182 @@
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API DDS PUT /v3/{project_id}/instances/{instance_id}/configurations
+// @API DDS GET /v3/{project_id}/instances
+// @API DDS GET /v3/{project_id}/jobs
+func ResourceDDSInstanceParametersModify() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDDSInstanceParametersModifyCreate,
+		ReadContext:   resourceDDSInstanceParametersModifyRead,
+		DeleteContext: resourceDDSInstanceParametersModifyDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"parameters": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"entity_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceDDSInstanceParametersModifyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.DdsV3Client(conf.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating DDS client: %s ", err)
+	}
+	instId := d.Get("instance_id").(string)
+	entityId := d.Get("entity_id").(string)
+	if entityId == "" {
+		entityId = instId
+	}
+	opt := buildDDSInstanceParametersModifyRequestBody(d.Get("parameters").([]interface{}), entityId)
+
+	// modify parameters
+	if ctx, err = modifyInstanceParameters(ctx, client, d.Timeout(schema.TimeoutCreate), instId, opt); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(instId)
+
+	return resourceDDSInstanceParametersModifyRead(ctx, d, meta)
+}
+
+func buildDDSInstanceParametersModifyRequestBody(params []interface{}, entityID string) map[string]interface{} {
+	values := make(map[string]string, len(params))
+	for _, v := range params {
+		key := v.(map[string]interface{})["name"].(string)
+		value := v.(map[string]interface{})["value"].(string)
+		values[key] = value
+	}
+
+	return map[string]interface{}{
+		"entity_id":        entityID,
+		"parameter_values": values,
+	}
+}
+
+func resourceDDSInstanceParametersModifyRead(ctx context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
```

Comment:
read method shoud be added, API:
https://support.huaweicloud.com/api-dds/dds_api_0108.html